newIDE/electron-app/app/main.js

```diff
@@ -268,6 +269,29 @@ app.on('ready', function() {
     }
   );
 
+  // LocalFilesystemWatcher events:
+  ipcMain.handle(
+    'local-filesystem-watcher-setup',
+    (event, folderPath, options) => {
+      const subscriptionId = setupWatcher(
+        folderPath,
+        (err, fileChangeEvents) => {
+          fileChangeEvents.forEach(fileChangeEvent =>
+            event.sender.send('project-file-changed', fileChangeEvent.path)
+          );
+        },
```

Comment:
Ubernit: The forEach could almost be inside newIDE/electron-app/app/LocalFilesystemWatcher.js. 
The main.js file job is to plug the things together (receive events, send events), but not to deal with the logic.